src/tau_clients/nsx_defender/__init__.py

```diff
@@ -998,13 +1001,21 @@ def get_result_artifact_names(
             if not report_types or any(x in report["report_versions"] for x in report_types):
                 temp = self.get_result(uuid, report_uuid=report["report_uuid"])
                 for item in temp.get("report", {}).get("analysis_metadata", []):
-                    if not metadata_types or item["metadata_type"] in metadata_types:
+                    if item["metadata_type"] in metadata_types or not metadata_types:
```

Comment:
Why did you swap the order inside if()? I would keep the original one to check if metadata_types empty or not first.

Reply:
Readability. And the second check is not really whether "metadata_types" is empty, but instead if "metadata_types" option was instead not specified at all, which is a corner case.